python/doc/examples/calibration/bayesian_calibration/plot_imh_python_distribution.py

```diff
@@ -1,100 +1,327 @@
+# Case 1: Computation of the normalization factor
+# -----------------------------------------------
+# The best thing to do is to compute the normalization factor thanks to the integration algorithms of the library.
+#
+# We show how to compute the normalization factor using a :class:`~openturns.GaussKronrod` quadrature formula.
+denom_fact = ot.GaussKronrod().integrate(f, range_PDF)[0]
+norm_fact = 1.0 / denom_fact
+print("normalization factor = ", norm_fact)
+
+# %%
+# Thus, we can define the exact PDF expression:
+#
+# .. math::
+#
+#   p(x) = \dfrac{f(x)}{\int_0^{2\pi} f(u)\, du}
+#
+exact_PDF = ot.LinearCombinationFunction([f], [norm_fact])
+
+# %%
+# Then we define a :class:`~openturns.PythonDistribution` which:
+#
+# - *computePDF* is computed from the exact expression,
+# - *computeCDF* is computed using an integration algorithm on the *computePDF*.
+#
+# Doing that way, we use the generic sampler of the distribution, based on the CDF inversion method,
+# implemented in the *getSample* method.
+
+
+class NewDistribution(ot.PythonDistribution):
+    def __init__(self):
+        super(NewDistribution, self).__init__(1)
+        self.PDF_ = exact_PDF
+        self.logPDF_ = ot.ComposedFunction(
+            ot.SymbolicFunction("x", "log(x)"), self.PDF_
+        )
+        self.rangeLow_ = 0.0
+        self.rangeUp_ = 2.0 * pi
+
+    def getRange(self):
+        return ot.Interval(self.rangeLow_, self.rangeUp_)
+
+    def computeLogPDF(self, X):
+        if X[0] < self.rangeLow_ or X[0] >= self.rangeUp_:
+            return -ot.SpecFunc.Infinity
+        return self.logPDF_(X)[0]
+
+    def computePDF(self, X):
+        if X[0] < self.rangeLow_ or X[0] >= self.rangeUp_:
+            return 0.0
+        return self.PDF_(X)[0]
+
+    def computeCDF(self, X):
+        if X[0] < self.rangeLow_:
+            return 0.0
+        if X[0] >= self.rangeUp_:
+            return 1.0
+        return ot.GaussLegendre([64]).integrate(
+            self.PDF_, ot.Interval(self.rangeLow_, X[0])
+        )[0]
+
+
+# %%
+# We create the new distribution that uses the generic sampler:
+newDist_generic = ot.Distribution(NewDistribution())
+
+# %%
+# We can draw the exact PDF and the PDF estimated from a sample generated by the Ratio of Uniforms algorithm.
+
+# %%
+size = 500
+sample = newDist_generic.getSample(size)
+
+ks_algo = ot.KernelSmoothing()
+ks_algo.setBoundaryCorrection(True)
+ks_algo.setLowerBound(lower_bound)
+ks_algo.setUpperBound(upper_bound)
+ks_pdf_GS = ks_algo.build(sample)
+
+g = ks_pdf_GS.drawPDF(-0.5, upper_bound * 1.1, 1001)
+draw_exact = exact_PDF.draw(lower_bound, upper_bound, 1001).getDrawable(0)
+draw_exact.setLineWidth(2)
+g.add(draw_exact)
+g.setLegends(["GS size = " + str(size), "exact pdf"])
+g.setLegendPosition("topright")
+g.setTitle("Generic sampler (GS)")
+g.setXTitle("x")
+view = otv.View(g)
+
+# %%
+# We can change the sampler of that new distribution by implementing the method *getSample* and *getRealization* as follows:
```

Comment:
here we could refer to Ratio of Uniforms:

"We can change the sampler of that new distribution relying on Ratio of Uniforms algorithm by implementing the method ..."

Reply:
yes.